src/Core/Checkout/Customer/SalesChannel/LoginAsCustomerRoute.php

```diff
@@ -0,0 +1,100 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Customer\SalesChannel;
+
+use Shopware\Core\Checkout\Customer\CustomerCollection;
+use Shopware\Core\Checkout\Customer\CustomerEntity;
+use Shopware\Core\Checkout\Customer\CustomerException;
+use Shopware\Core\Checkout\Customer\Event\CustomerBeforeLoginEvent;
+use Shopware\Core\Checkout\Customer\Event\CustomerLoginEvent;
+use Shopware\Core\Checkout\Customer\Exception\CustomerNotFoundByIdException;
+use Shopware\Core\Checkout\Customer\LoginAsCustomerTokenGenerator;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Plugin\Exception\DecorationPatternException;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\System\SalesChannel\Context\CartRestorer;
+use Shopware\Core\System\SalesChannel\ContextTokenResponse;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+#[Route(defaults: ['_routeScope' => ['store-api'], '_contextTokenRequired' => false])]
+#[Package('checkout')]
+class LoginAsCustomerRoute extends AbstractLoginAsCustomerRoute
+{
+    public const CUSTOMER_ID = 'customerId';
+
+    public const SALES_CHANNEL_ID = 'salesChannelId';
+
+    public const TOKEN = 'token';
+
+    /**
+     * @internal
+     *
+     * @param EntityRepository<CustomerCollection> $customerRepository
+     */
+    public function __construct(
+        private readonly EventDispatcherInterface $eventDispatcher,
+        private readonly EntityRepository $customerRepository,
+        private readonly CartRestorer $restorer,
+        private readonly LoginAsCustomerTokenGenerator $tokenGenerator
+    ) {
+    }
+
+    public function getDecorated(): AbstractLoginAsCustomerRoute
+    {
+        throw new DecorationPatternException(self::class);
+    }
+
+    #[Route(path: '/store-api/account/login/customer', name: 'store-api.account.login-as-customer', methods: ['POST'])]
+    public function loginAsCustomer(RequestDataBag $data, SalesChannelContext $context): ContextTokenResponse
+    {
+        // TODO: find better way to handle this
```

Comment:
You can use here the DataValidator which uses under the hood Symfony Validator

https://github.com/shopware/shopware/blob/66e20695afc4e6fd7c48ac587e0acc81f9d04486/src/Core/Content/Newsletter/SalesChannel/NewsletterUnsubscribeRoute.php#L91-L92